### ISSUE TITLE
A usys with a small length unit (e.g. si) silently degrades the SOM ordering via float32 overflow

### ISSUE THREAD

**Issue by nstarman** [GalacticDynamics/phasecurvefit#86]
Found during an independent audit of #80/#56. **Pre-existing on `main`** (the overflow is in `init_prototypes`' PCA fallback, and the strip-into-`usys` path predates both PRs) — filing separately rather than widening them.

## A `usys` whose length unit is much smaller than the data silently degrades the ordering

The Quantity dispatches strip into `usys`, so the *numeric magnitude* reaching the float32 core is the caller's unit-system choice. Choose SI for galactic data and the numbers become ~1e20, which overflows float32 in the PCA Gram matrix — with **no exception, no warning and no NaN in the output**.

### Reproduction

Identical physics, only `usys` differs; a 120-point half-circle of radius 10 kpc:

```python
r = pcf.orderers.SOMOrderer(n_prototypes=12).order(q, p, metadata=StateMetadata(usys=usys))
```

| `usys` | \|rho\| vs the true curve parameter | chord max | n_visited |
|---|---|---|---|
| `galactic` | **1.0000** | 28.55 | 120 |
| `si` | **0.7501** | 2.621e+05 | 120 |

`n_visited` still reports 120/120, so nothing downstream can tell the ordering was degraded.

### Root cause

`init_prototypes`' PCA fallback (`_src/som.py`, `centered.T @ centered`) squares the positions in float32:

```
position magnitude 1e17                   Gram finite=True   max=6.05e+37
position magnitude 1e18                   Gram finite=False  max=inf
position magnitude 3.086e19 (1 kpc in m)  Gram finite=False  max=inf
```

float32 tops out at 3.4e38, so the Gram overflows above a position magnitude of ~1e18 — and **1 kpc = 3.086e19 m**, which puts ordinary galactic data in SI over the edge before any user error.

`MSTOrderer` is unaffected (host-side float64). This bites the JAX orderers.

### Why the tests don't catch it

`tests/test_orderers_unxt.py` uses `usys=galactic` throughout, whose length unit is `kpc` — exactly the data's own unit — so nothing exercises a magnitude change. The scale-invariance test added in #80 (`test_chord_is_invariant_to_the_position_scale`) sweeps `1.0, 1e-6, 1e-8`, i.e. **downward only**; the failure is upward. Adding `1e18` to that loop turns this into a failing test.

### Suggested direction

Options, roughly in order of appeal:

1. **Normalise internally.** Centre and scale positions before the Gram (and rescale the prototypes back), which makes `init_prototypes` scale-free rather than merely lucky in kpc.
2. **Compute the Gram in float64** on host, or via `jnp.linalg.svd` on the centred matrix rather than forming `XᵀX` at all — the SVD route avoids the squaring entirely and is the numerically standard way to get a principal axis.
3. **Validate at the boundary** — reject or warn when the stripped magnitude is within a decade or two of the float32 ceiling. Weakest option alone, but a good complement.

Option 2 is probably the cleanest: it removes the squaring that causes the overflow, rather than working around it.
